### PR TITLE
Improved Sleeper - READY FOR MERGE

### PR DESCRIPTION
**Pull request by Surrealaser**
-Now has a cryostasis function; particularly useful for overdosed and rapidly deteriorating patients.

-Dialysis works much faster.

-No delay on climbing/putting people in.

-Can inject people who are at negative health.

-Examining lets you know who is inside and which processes are running as well as their medical reports.

-Health analyzers will scan the Sleeper's occupant when used on the Sleeper.

### CODE REVIEW

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -398,31 +474,30 @@
 		to_chat(usr, "\blue <B>The sleeper is already occupied!</B>")
 		return
 
-	visible_message("[usr] starts climbing into the sleeper.", 3)
+	visible_message("[usr] climbs into the sleeper.", 3)
```
[the remark this answers is not in the document]
Comment (Surrealaser) [reply code/game/machinery/Sleeper.dm]: There is no delay anymore; this action is now instant.

**Comment by Rohesie** [comment]
This is dependent on #377 . After that one is in, it can be tested.

File: code/game/machinery/Sleeper.dm [hunk 2]
```diff
@@ -190,6 +195,55 @@
...
 
+/obj/machinery/sleeper/Dispose()
+	occupant.in_stasis = FALSE //clean up; end stasis; remove from processing
+	occupant = null
+	processing_objects.Remove(src)
+	stop_processing()
+	return ..()
+
+/obj/machinery/sleeper/examine(mob/living/user)
+	..()
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]: `. = ..()`

File: code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -190,6 +195,55 @@
...
+	if(!occupant) //Allows us to reference medical files/scan reports for cryo via examination.
+		return
+	if(!ishuman(occupant))
+		return
+	var/active = ""
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]: Can you name this `active_feedback` or something that doesn't suggest it's a boolean?

File: code/game/machinery/Sleeper.dm [hunk 4]
```diff
@@ -201,13 +255,18 @@
...
 		return
 
-	if(filtering > 0)
+	if(filtering)
 		if(beaker)
 			if(beaker.reagents.total_volume < beaker.reagents.maximum_volume)
-				for(var/datum/reagent/x in occupant.reagents.reagent_list)
-					occupant.reagents.trans_to(beaker, 3)
+				for(var/datum/reagent/x in src.occupant.reagents.reagent_list)
+					src.occupant.reagents.trans_to(beaker, 10)
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]: Is this intended or the byproduct of not rebasing?

File: code/game/machinery/Sleeper.dm [hunk 5]
```diff
@@ -225,6 +284,10 @@
 			to_chat(user, "<span class='warning'>The sleeper has a beaker already.</span>")
 			return
 
+	else if(istype(W, /obj/item/device/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
+		var/obj/item/device/healthanalyzer/J = W
+		J.attack(occupant, user)
+
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]: A `return` here?

File: code/game/machinery/Sleeper.dm [hunk 6]
```diff
@@ -236,26 +299,21 @@
 			to_chat(user, "<span class='notice'>The sleeper is already occupied!</span>")
 			return
 
-		visible_message("[user] starts putting [G.grabbed_thing] into the sleeper.", 3)
+		visible_message("[user] puts [G.grabbed_thing] into the sleeper.", 3)
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]: Add this message after the `grabbed_thing` has be actually put in.

File: code/game/machinery/Sleeper.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rohesie) [reply code/game/machinery/Sleeper.dm]: Yes, but `forceMove` or something else could fail for some reason. Add feedback of what has happened after the deed is done, not before.

File: code/game/machinery/Sleeper.dm (hunk 4 above)
Comment (Surrealaser) [reply code/game/machinery/Sleeper.dm]: Intended.

File: code/game/machinery/Sleeper.dm [hunk 7]
```diff
@@ -385,29 +459,22 @@
...
-		occupant = usr
-		start_processing()
-		connected.start_processing()
-		icon_state = "sleeper_1"
-		if(orient == "RIGHT")
-			icon_state = "sleeper_1-r"
+	user.stop_pulling()
+	if(!user.forceMove(src))
+		return
+	visible_message("[usr] climbs into the sleeper.", 3)
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]:
```suggestion
	visible_message("[user] climbs into the sleeper.", 3)
```

File: code/game/machinery/Sleeper.dm [hunk 8]
```diff
@@ -385,29 +459,22 @@
 		to_chat(usr, "<span class='notice'>The sleeper is already occupied!</span>")
 		return
 
-	visible_message("[user] starts climbing into the sleeper.", 3)
-	if(user.pulledby)
-		if(ismob(user.pulledby))
-			var/mob/grabmob = user.pulledby
+	if(usr.pulledby)
+		if(ismob(usr.pulledby))
+			var/mob/grabmob = usr.pulledby
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]:
```suggestion
			var/mob/grabmob = user.pulledby
```

File: code/game/machinery/Sleeper.dm [hunk 9]
```diff
@@ -385,29 +459,22 @@
 		to_chat(usr, "<span class='notice'>The sleeper is already occupied!</span>")
 		return
 
-	visible_message("[user] starts climbing into the sleeper.", 3)
-	if(user.pulledby)
-		if(ismob(user.pulledby))
-			var/mob/grabmob = user.pulledby
+	if(usr.pulledby)
+		if(ismob(usr.pulledby))
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]:
```suggestion
		if(ismob(user.pulledby))
```

Comment (Rohesie) [reply code/game/machinery/Sleeper.dm]: Also, unnest this.

File: code/game/machinery/Sleeper.dm [hunk 10]
```diff
@@ -385,29 +459,22 @@
 		to_chat(usr, "<span class='notice'>The sleeper is already occupied!</span>")
 		return
 
-	visible_message("[user] starts climbing into the sleeper.", 3)
-	if(user.pulledby)
-		if(ismob(user.pulledby))
-			var/mob/grabmob = user.pulledby
+	if(usr.pulledby)
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]:
```suggestion
	if(user.pulledby)
```

File: code/game/machinery/Sleeper.dm [hunk 11]
```diff
@@ -96,6 +96,10 @@
 					else
 						dat += "<HR><A href='?src=\ref[src];togglefilter=1'>Start Dialysis</A><BR>"
 						dat += "Output Beaker has [connected.beaker.reagents.maximum_volume - connected.beaker.reagents.total_volume] units of free space remaining<BR><HR>"
+					if(src.connected.stasis)
```
Comment (Rohesie) [inline code/game/machinery/Sleeper.dm]:
```suggestion
					if(connected.stasis)
```

**Pull request merged**